### PR TITLE
#1098: [Typo] TextOverflow (closes #1098)

### LINKED ISSUE #1098: [Typo] TextOverflow
[Project card](https://github.com/buildo/react-components/projects/1?fullscreen=true#card-4470981)

← #1034

## requirements
{describe the new feature}

## specs
{optional: describe technical specs to implement this feature, if not obvious}

## misc
{optional: other useful info}

### PR DESCRIPTION
**Pull request by gabro**
Closes #1098

### tests performed
Tested locally on showroom. It works as expected.

### CODE REVIEW

File: src/Button/Button.tsx [hunk 1]
```diff
@@ -17,14 +15,7 @@ export namespace ButtonProps {
...
 
-// TODO: TextOverflow should accept a popoverProps
-export type TextOverflowCompatibleComponent = React.ComponentClass<{
-  label?: string | number,
-  popover?: ObjectOverwrite<PopoverProps.Popover, {
-    position?: PopoverProps.Position,
-    content?: void & string
-  }>
-}>
+export type TextOverflowCompatibleComponent = React.ComponentClass<TextOverflowProps>;
```
Comment (gabro) [inline src/Button/Button.tsx]: Double check this, but it seems correct and it makes it more consistent.

File: src/TextOverflow/TextOverflow.tsx [hunk 2]
```diff
@@ -21,28 +23,53 @@ export const Props = {
...
 @props(Props, { strict: false })
-export default class TextOverflow extends React.Component {
+export default class TextOverflow extends React.Component<TextOverflowProps, TextOverflowState> {
 
-  static defaultProps = {
-    delayWhenLazy: 100
-  }
+  static defaultProps: TextOverflowDefaultProps = {
+    delayWhenLazy: 100,
+    lazy: false
```
Comment (gabro) [inline src/TextOverflow/TextOverflow.tsx]: I explicitly set `lazy` to `false`

File: src/TextOverflow/TextOverflow.tsx [hunk 3]
```diff
@@ -21,28 +23,53 @@ export const Props = {
...
+  };
 
-  state = { isOverflowing: false };
+  state = { isOverflowing: false, isHovering: false };
```
Comment (gabro) [inline src/TextOverflow/TextOverflow.tsx]: same here, I explicitly initialized `isHovering` to `false.`

File: src/TextOverflow/TextOverflow.tsx [hunk 4]
```diff
@@ -56,7 +83,7 @@ export default class TextOverflow extends React.Component {
     warn(() => {
       const node = ReactDOM.findDOMNode(this.refs.text);
       if (node) {
-        const styleNode = node.parentNode.parentNode;
+        const styleNode = node.parentElement!.parentElement!;
```
Comment (gabro) [inline src/TextOverflow/TextOverflow.tsx]: As usual, `!` is safe because only `window` has no parent element

**Comment by FrancescoCioria** [comment]
@gabro have a look at my last commit

**Comment by FrancescoCioria** [comment]
@gabro `git push` had failed 👅

**Comment by gabro** [comment]
Looks good!

**Pull request merged**